apps/mark/backend/src/app.ts

```diff
@@ -1,18 +1,106 @@
+import * as grout from '@votingworks/grout';
+import { BallotStyleId, Optional, PrecinctId } from '@votingworks/types';
+import { ScannerReportData, ScannerReportDataSchema } from '@votingworks/utils';
+
 import { getMachineConfig } from './machine_config';
 
-function buildApi() {
+function buildApi(auth: InsertedSmartCardAuthApi) {
   return grout.createApi({
     getMachineConfig,
+
+    // TODO: Once election definition has been moved to the backend, no longer require the frontend
+    // to provide election hash to this and other methods that use the auth lib
+    getAuthStatus({ electionHash }: { electionHash?: string }) {
+      return auth.getAuthStatus({ electionHash });
+    },
+
+    checkPin({ electionHash, pin }: { electionHash?: string; pin: string }) {
+      return auth.checkPin({ electionHash }, { pin });
+    },
+
+    startCardlessVoterSession({
+      electionHash,
+      ballotStyleId,
+      precinctId,
+    }: {
+      electionHash?: string;
+      ballotStyleId: BallotStyleId;
+      precinctId: PrecinctId;
+    }) {
+      return auth.startCardlessVoterSession(
+        { electionHash },
+        { ballotStyleId, precinctId }
+      );
+    },
+
+    endCardlessVoterSession({ electionHash }: { electionHash?: string }) {
+      return auth.endCardlessVoterSession({ electionHash });
+    },
+
+    async readElectionDefinitionFromCard({
+      electionHash,
+    }: {
+      electionHash?: string;
+    }): Promise<Result<Optional<string>, Error>> {
+      const authStatus = await auth.getAuthStatus({ electionHash });
+      if (authStatus.status !== 'logged_in') {
+        return err(new Error('User is not logged in'));
+      }
+      if (authStatus.user.role !== 'election_manager') {
+        return err(new Error('User is not an election manager'));
+      }
+
+      return await auth.readCardDataAsString({ electionHash });
+    },
+
+    async readScannerReportDataFromCard({
+      electionHash,
+    }: {
+      electionHash?: string;
+    }): Promise<
+      Result<Optional<ScannerReportData>, SyntaxError | z.ZodError | Error>
+    > {
+      const authStatus = await auth.getAuthStatus({ electionHash });
+      if (authStatus.status !== 'logged_in') {
+        return err(new Error('User is not logged in'));
+      }
+      if (authStatus.user.role !== 'poll_worker') {
+        return err(new Error('User is not a poll worker'));
+      }
+
+      return await auth.readCardData(
+        { electionHash },
+        { schema: ScannerReportDataSchema }
+      );
+    },
+
+    async clearScannerReportDataFromCard({
+      electionHash,
+    }: {
+      electionHash?: string;
+    }): Promise<Result<void, Error>> {
+      const authStatus = await auth.getAuthStatus({ electionHash });
+      if (authStatus.status !== 'logged_in') {
+        return err(new Error('User is not logged in'));
+      }
+      if (authStatus.user.role !== 'poll_worker') {
+        return err(new Error('User is not a poll worker'));
+      }
```

Comment:
I like this model of checking role-based permissions in the API. That meshes with general patterns I've seen for authentication vs authorization.

I wonder if the `logged_in` check should be part of the auth lib or stay here in the API. I don't have a strong opinion.

Reply:
Hmm interesting question. I do like having these checks exist together as opposed to split across the codebase. Also, once we start gating the backend more consistently, we'll have API methods that don't read or write card data but that also need to check if the user is logged in. For those, we'll have to put the logged-in check here in the API layer, so might as well do so consistently

Reply:
Sounds good!